src/app/index.ts

```diff
@@ -0,0 +1,1003 @@
+              </div>
+              </td>
+              <td>
+                <button class="btn-reset cart-item__delete" data-id="${item.id
+                }" type="button">
+                  
+                </button>
+              </td>
+            </tr>
+          `;
+        });
+
+        (document.querySelector(".make-order .price") as Element).innerHTML =
+            currentUser.cart.totalSum;
+    } else {
+        (cartBody as Element).innerHTML = afterOrder ?
+            `
+      <tr>
+        <td class="fade"><p style="padding: 30px 14px 80px;
+        font-size: 36px;">Thank you for your order</p></td>
+      </tr>
+        ` :
+            `
+        <tr>
+          <td class="fade"><p style="padding: 30px 14px 80px;
+          font-size: 36px;">Your cart is empty</p></td>
+        </tr>
+        `;
+        afterOrder = false;
+        const price = document.querySelector(".make-order .price");
+        const priceHeader = document.querySelector(".header-order .price");
+        (price as Element).innerHTML = "";
+        (document.querySelector(".make-order") as HTMLElement).style.display = "none";
+    }
+
+    const itemDelete = document.querySelectorAll(".cart-item__delete");
+
+    itemDelete.forEach((item: any) => {
+        item.addEventListener("click", (e: any) => {
+            e.stopPropagation();
+            currentUser.cart.removeProduct(item.dataset.id);
+            setCardCount();
+            renderCart();
+        });
+    });
+    updateSearchPatams();
+    saveToLS(currentUser);
+}
+
+(cartBody as Element).addEventListener("click", counterControl);
+
+function counterControl(e: any) {
+    e.stopPropagation();
+    const counter = e.target.closest(".counter");
+
+    if (counter) {
+        const productId = counter.dataset.id;
+        const counterInput = counter.querySelector(".counter__input");
+        const incr = counter.querySelector(".counter__incr");
+        const dcr = counter.querySelector(".counter__dcr");
+        const productStock = findProduct(productId).amount;
+
+        if (dcr.contains(e.target)) {
+            if (counterInput.value > 1 && counterInput.value <= productStock) {
+                counterInput.value--;
+                incr.disabled = false;
+                currentUser.cart.changeQuantity(productId, counterInput.value);
+            } else {
+                currentUser.cart.removeProduct(productId);
+            }
+            renderCart();
+        } else if (incr.contains(e.target)) {
+            if (counterInput.value < productStock) {
+                counterInput.value++;
+                currentUser.cart.changeQuantity(productId, counterInput.value);
+            }
+            if (counterInput.value == productStock) {
+                incr.disabled = true;
+            }
+            renderCart();
+        }
+        setCardCount();
+    }
+}
+
+(cartBody as Element).addEventListener("keypress", (e) => {
+    let target = e.target as HTMLElement;
```

Comment:
I think if the actions are taken out of the listeners into separate functions, the code will be more readable and clean